app/models/post.rb

```diff
@@ -0,0 +1,18 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :likes
+  has_many :comments
+
+  after_create :update_user_posts_counter
+  after_destroy :update_user_posts_counter
+
+  # updates the posts counter for a user.
+  def update_user_posts_counter
+    author.update(posts_counter: author.posts.count)
+  end
+
+  # returns the 5 most recent comments for a given post.
+  def self.five_most_recent_comments(post_id)
+    Comment.where(post_id:).order(created_at: :desc).limit(5)
+  end
```

Comment:
- [ ] [OPTIONAL] I would suggest that you take full advantage of the has_many relationship between the post and its comments like this 👇
```ruby
def five_most_recent_comments
  comments.order('created_at Desc').limit(5)
end

```